src/core/Normalization.helper.js

```diff
@@ -0,0 +1,35 @@
+/* eslint-disable no-plusplus */
+/* eslint no-await-in-loop: 0 */
+export default class NormalizationHelper {
+    /**
+     *
+     * @param {String|Number} value  - The value that will be normalized
+     * @param {Boolean} fullUnit - This is by default on true. If you put this on false, the method will not do the multiplication
+     * @returns {Number}
+     */
+    static normalizeGoalValue (value, fullUnit) {
+
+        if (fullUnit !== false) {
+            fullUnit = true;
+        }
+
+        let normalized = parseFloat(value);
+        if (isNaN(normalized)) {
+            return 0;
```

Comment:
@robertdumitrescu MAINTAINABILITY - `0` or `null` - they could mean slightly different things.

Reply:
I am happy to go with null, but currently, the goalValue if is not found or is corrupt, we put 0 instead. 

```
if (transaction) {
            event.goalValue = parseFloat(transaction.bundle.contractValueCable);
          } else {
            event.goalValue = 0;
          }
```

Reply:
Fixed according to the private discussion. Therefore corrupt values will be nullified.